src/cow.c

```diff
@@ -61,7 +61,8 @@ int cow_cp(const char *path, int branch_ro, int branch_rw, bool copy_dir) {
 	DBG("%s\n", path);
 
 	// create the path to the file
-	path_create_cutlast_cow(path, branch_ro, branch_rw);
+	int res = path_create_cutlast_cow(path, branch_ro, branch_rw);
+	if (res != 0) RETURN(res);
```

Comment:
Previously the execution went on to try to copy the file even if creating its path failed.